File: SP-API/Domain/Order.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Order
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string OrderCode { get; set; }
        public decimal TotalAmount { get; set; }
        public decimal TotalBillDiscount { get; set; }
        public decimal TotalNetAmount { get; set; }
        public bool IsComplete { get; set; }
        public DateTime? CompletedDate { get; set; }
        public DateTime OrderedDate { get; set; }
        public bool IsEdit { get; set; }
        public DateTime? EditedDate { get; set; }
        public int EditedUserId { get; set; }
        public bool IsCancel { get; set; }
        public DateTime? CanceledDate { get; set; }
        public int CanceledUserId { get; set; }
        public string CanceledReason { get; set; }
        public virtual ICollection<OrderItemBatch> OrderItemBatches { get; set; }
        public virtual UserDetail UserDetail { get; set; }
    }
}

File: SP-Client/src/app/_models/order.ts
import { OrderItemBatch } from "./orderItemBatch";

export class Order {
  id: number;
  salesRepName: string;
  orderCode: string;
  totalAmount: number;
  totalBillDiscount: number;
  totalDiscountAmount: number;
  totalNetAmount: number;
  isComplete: boolean;
  completedDate: Date;
  orderedDate: any;
  isEdit: boolean;
  editedDate: Date;
  editedUserId: number;
  editedUserName: string;
  isCancel: boolean;
  canceledDate: Date;
  canceledUserId: number;
  canceledUserName: string;
  canceledReason: string;
  loginEmail: string;
  orderItemBatches: OrderItemBatch[];

  constructor(order) {
    this.id = order.id;
    this.salesRepName = order.salesRepName;
    this.orderCode = order.orderCode;
    this.totalAmount = order.totalAmount;
    this.totalBillDiscount = order.totalBillDiscount;
    this.totalDiscountAmount = order.totalDiscountAmount;
    this.totalNetAmount = order.totalNetAmount;
    this.isComplete = order.isComplete;
    this.completedDate = order.completedDate;
    this.orderedDate = order.orderedDate;
    this.isEdit = order.isEdit;
    this.editedDate = order.editedDate;
    this.editedUserId = order.editedUserId;
    this.isCancel = order.isCancel;
    this.canceledDate = order.canceledDate;
    this.canceledUserId = order.canceledUserId;
    this.canceledReason = order.canceledReason;
    this.orderItemBatches = order.orderItemBatches;
    this.editedUserName = order.editedUserName;
    this.canceledUserName = order.canceledUserName;
  }
}
